File: CaysarCipher.Py/substitutionCipher.py
import random

class SubstittutionCipher():
    def generate_key():
        domain = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        cletters = list(domain)
        key = {}
        for c in domain:
            key[c] = cletters.pop(random.randint(0, len(cletters) - 1))
        return key
    
    def get_decryption_key(key):
        dkey = {}
        for c in key:
            dkey[key[c]] = c
        return dkey

    def encrypt(key, message):
        encryped_string = ""
        for c in message:
            if c in key:
                encryped_string += key[c]
            else:
                encryped_string += c
        return encryped_string
        # for i in range(len(value)):
        #     if(key.get(value[i])):
        #         encryped_string = encryped_string.__add__(key[value[i]])
        #     else: 
        #         encryped_string = encryped_string.__add__(value[i])

text = "CAR"
key = SubstittutionCipher.generate_key()
print(key)
print(SubstittutionCipher.encrypt(key, text))

File: SubstitutionCipher.CSharp/SubstitutionCipher.cs
using System.Text;

namespace Substitution
{
    public class SubstitutionCipher
    {
        public static Dictionary<char, char> GenerateKey()
        {
            var keys = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
            var domain = new List<char>(keys);
            Random random = new Random();
            var key = new Dictionary<char, char>();

            foreach (char c in keys)
            {
                int index = random.Next(domain.Count);
                key[c] = domain.ElementAt(index);
                domain.RemoveAt(index);
            }

            return key;
        }

        public static Dictionary<char, char> GetDecryptKey(Dictionary<char, char> key)
        {
            var dKey = new Dictionary<char, char>();
            foreach (KeyValuePair<char, char> item in key)
            {
                dKey[item.Value] = item.Key;
            }
            return dKey;
        }

        public static string Encrypt(Dictionary<char, char> key, string value)
        {
            StringBuilder encrypedString = new();
            for (var i = 0; i < value.Length; i++)
            {
                if (key.ContainsKey(value[i]))
                {
                    encrypedString.Append(key[value[i]]);
                }
                else // any another char like * _ space ...
                {
                    encrypedString.Append(value[i]);
                }
            }
            return encrypedString.ToString();
        }
    }
}
